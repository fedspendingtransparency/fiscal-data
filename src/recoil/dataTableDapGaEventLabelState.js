import { atom } from 'recoil';

export const dataTableDapGaEventLabelState = atom({
  key: 'dataTableDapGaEventLabelState',
  default: '',
});
