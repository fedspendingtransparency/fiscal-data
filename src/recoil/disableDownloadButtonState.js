import { atom } from 'recoil';

export const disableDownloadButtonState = atom({
  key: 'disableDownloadButtonState',
  default: false,
});
