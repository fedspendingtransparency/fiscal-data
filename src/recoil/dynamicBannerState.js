import { atom, selector } from "recoil";
import { apiPrefix, basicFetch } from "../utils/api-utils";

export const dynamicBannerLastCachedState = atom({
  key: 'dynamicBannerLastCachedState',
  default: 0,
});

export const dynamicBannerState = selector({
  key: 'dynamicBannerState',
  get: async () => {
    const announcementsEndpoint = 'v1/reference/fiscal_data/announcements';
    const response = await basicFetch(`${apiPrefix}${announcementsEndpoint}`);
    const timeStamp = Date.now();
    return {
      payload: response.data,
      timeStamp: timeStamp,
    };
  },
});
