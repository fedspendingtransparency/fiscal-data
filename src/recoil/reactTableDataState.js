import { atom } from 'recoil';

export const reactTablePageState = atom({
  key: 'reactTablePage',
  default: 0,
});

export const reactTableDePaginatedDataState = atom({
  key: 'reactTableDePaginatedDataState',
  default: null,
});
