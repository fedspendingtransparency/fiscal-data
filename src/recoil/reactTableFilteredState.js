import { atom } from 'recoil';

export const reactTableFilteredDateRangeState = atom({
  key: 'reactTableFilteredDateRangeState',
  default: null,
});

export const reactTableSortingState = atom({
  key: 'reactTableSortingState',
  default: [],
});
