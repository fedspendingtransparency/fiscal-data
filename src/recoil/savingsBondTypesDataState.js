import { selector, atom } from 'recoil';
import { apiPrefix, basicFetch } from '../utils/api-utils';

export const savingsBondTypesLastCachedState = atom({
  key: 'savingsBondTypesLastCachedState ',
  default: 0,
});

export const savingsBondTypesData = selector({
  key: 'savingsBondTypesData',
  get: async ({ get }) => {
    const numOfSavingsBonds = 'v1/accounting/od/securities_sales?filter=security_type_desc:eq:Savings%20Bond';
    const response = await basicFetch(`${apiPrefix}${numOfSavingsBonds}&page[size]=1`).then(metaRes => {
      if (metaRes.meta && typeof metaRes.meta['total-pages'] !== 'undefined') {
        const pageSize = metaRes.meta['total-pages'];
        return basicFetch(`${apiPrefix}${numOfSavingsBonds}&page[size]=${pageSize}`);
      }
    });
    const timeStamp = Date.now();
    return {
      payload: response.data,
      timeStamp: timeStamp,
    };
  },
});
