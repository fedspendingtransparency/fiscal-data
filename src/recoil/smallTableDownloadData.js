import { atom } from 'recoil';

export const smallTableDownloadDataCSV = atom({
  key: 'smallTableDownloadDataCSV',
  default: [],
});

export const smallTableDownloadDataJSON = atom({
  key: 'smallTableDownloadDataJSON',
  default: '',
});

export const smallTableDownloadDataXML = atom({
  key: 'smallTableDownloadDataXML',
  default: {},
});

export const tableRowLengthState = atom({
  key: 'tableRowLengthState',
  default: null,
});
