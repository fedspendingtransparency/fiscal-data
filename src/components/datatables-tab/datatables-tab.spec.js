import React from 'react';
import renderer from 'react-test-renderer';
import DtgTable from '../dtg-table/dtg-table';
import DataTablesTab from './datatables-tab';
import { RecoilRoot } from 'recoil';

describe('DataTablesTab', () => {
  const mockData = [
    {
      tableName: 'table 1',
      tableDescription: 'table 1 description',
      rowCount: '9999',
      rowDefinition: 'this row does this',
    },
    {
      tableName: 'table 2',
      tableDescription: 'table 2 description',
      rowCount: '111',
      rowDefinition: 'that row does that',
    },
  ];
  let component = renderer.create();
  let instance;
  beforeAll(() => {
    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <DataTablesTab apis={mockData} />
        </RecoilRoot>
      );
    });
    instance = component.root;
  });

  it('should pass along its data array to the dtgTable component', () => {
    const tableData = instance.findByType(DtgTable).props.tableProps.data;
    expect(tableData).toMatchSnapshot();
  });
  it('sets aria-label to [dataset name] data tables', () => {
    const name = 'test-dataset';
    const newComponent = renderer.create();
    renderer.act(() => {
      newComponent.update(
        <RecoilRoot>
          <DataTablesTab apis={mockData} datasetName={name} />
        </RecoilRoot>
      );
    });
    const updated = newComponent.root;
    const table = updated.findByType('table');
    expect(table.props['aria-label']).toBe(`${name} data tables`);
  });
});
