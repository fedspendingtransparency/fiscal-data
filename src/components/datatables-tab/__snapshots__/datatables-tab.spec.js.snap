// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTablesTab should pass along its data array to the dtgTable component 1`] = `
[
  {
    "description": "table 1 description",
    "name": "table 1",
    "rowCount": "9999",
    "rowDescription": "this row does this",
  },
  {
    "description": "table 2 description",
    "name": "table 2",
    "rowCount": "111",
    "rowDescription": "that row does that",
  },
]
`;
